src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java

```diff
@@ -1,31 +1,121 @@
 package com.endava.internship.mocking.service;
 
+import com.endava.internship.mocking.model.Payment;
+import com.endava.internship.mocking.model.Status;
+import com.endava.internship.mocking.model.User;
+import com.endava.internship.mocking.repository.InMemPaymentRepository;
+import com.endava.internship.mocking.repository.InMemUserRepository;
+import com.endava.internship.mocking.repository.PaymentRepository;
+import com.endava.internship.mocking.repository.UserRepository;
+import static org.assertj.core.api.Assertions.*;
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+
+import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.Mockito;
 import org.mockito.junit.jupiter.MockitoExtension;
 
+import java.util.*;
+
 @ExtendWith(MockitoExtension.class)
 class PaymentServiceTest {
-
+    @Mock
+    ValidationService basicValidationService;
+    @Mock
+    UserRepository userRepository;
+    @Mock
+    PaymentRepository paymentRepository;
+    PaymentService paymentService;
 
     @BeforeEach
     void setUp() {
-
+        basicValidationService = Mockito.mock(BasicValidationService.class);
+        userRepository = Mockito.mock(InMemUserRepository.class);
+        paymentRepository = Mockito.mock(InMemPaymentRepository.class);
+        paymentService = new PaymentService(userRepository,paymentRepository,basicValidationService);
     }
+    @Test
+    void createPayment_whenUserNotExists_ShouldThrowNoSuchElementException() {
+        Integer userId = 7;
+        Double amount = 500d;
+
+        Assertions.assertThrows(NoSuchElementException.class, () -> paymentService.createPayment(userId,amount));
 
```

Comment:
add verifyNoInteractions(userRepository) here and for other tests where mocks are not used

Reply:
fixed